### ISSUE TITLE
Schedule Design question 

### ISSUE THREAD

**Issue by fderyckel** [timnon/pyschedule#13]
Hi @timnon 
I'm posting this here as I'm not sure where else to ask;
I have a design question:
Can we have 2 type of tasks but constraints on only one of the type.  
Here is the scenario.  
Let's say a teacher can perform up to 6 teaching task but no more than 3 in a row on a 8 periods time.  That's ok to model.  
Now we need to add a department meeting task or lunch for both R1 and R2
So this is OK
Teaching, Teaching, Meeting, Teaching,    , Teaching, Teaching, Teaching

And this is OK
Teaching, Teaching, Teaching, Meeting, Teaching,      , Teaching

But this is NOT OK:
Teaching, Teaching, Teaching, Teaching,    , Meeting, Teaching, Teaching

Basically, how can we model:
for any 4 consecutive periods: 
    it should contain either 
        no more than 3 tasks or
        it should include the task "meeting/lunch"?

**Comment by timnon** [comment]
Can you create a constraint for every 4 consecutive periods where a meeting has a large negative value? So whenever you have one, there can be as many lectures with posititve values as you want? Does this make sense?

If you have a complete description of your task with example input data, please post, I would be interested to have more practical examples:)

**Comment by fderyckel** [comment]
Hi Timnon??,

Greetings from Lusaka.
 
First I want to thank you for sharing your library.  It is an absolutely wonderful piece of software.  I managed to do the whole schedule for our school and the schedule is much better than the one we currently have where over the last few years we had to flew a guy from the USA who used quite an expensive school management system. 

I'm currently living in Zambia where I teach math at the American International School of Lusaka.  

I'm attaching the Jupyter notebook I've created.  [here](https://drive.google.com/open?id=0ByHtvgo2NGDMcHFXd3MtWHdmUzA)  As you can see I'm not a professional programmer and there are many ways we could have done the code shorter.  I'm still learning from your examples (but they are so concise that I really have to go step by step).  I use R almost daily, so going to Python is quite a jump ;-)

The constraints where (most teachers have to teach 5 periods / over a 8 periods cycle).
1 day is 4 periods.  Ideally no teachers should teach a whole day (although teaching 3 periods and 1 period meeting is OK)
Each dpt. should meet at least one time for collaborative planning.  


Any way, I'm very excited by what we have achieved thanks to you sharing Pyschedule.

Best regards, 

François  f.deryckel@gmail.com

**Comment by timnon** [comment]
Hi François,
very nice, love this story!  So far i have not received much feedback about practical applications, and so the development has paused for some months, maybe it makes sense to put in more efforts. If that is ok for you, i would post your notebook as an example in project pyschedule-notebooks?
Cheers,
Tim

**Comment by fderyckel** [comment]
@timnon ,

Yes, Tim that is absolutely fine.  Thanks for the advice earlier.
